designs/air-gapped.md

```diff
@@ -0,0 +1,97 @@
+
+### Tenets
+
+* ****Simple:**** minimal interface, minimum number of manual steps.
+
+### Goals and Objectives
+
+As a EKS-A cluster administrator I want to:
+
+* Manage clusters in an environment without Internet connection.
+* Populate my environment with the necessary dependencies without having to worry about what those dependencies are and with minimal effort.
+* Interact with EKS-A clusters in air-gapped environments in the same way I interact with any other EKS-A cluster.
+
+### Statement of Scope
+
+**In scope**
+* Download dependencies and package them.
+* Populate disconnected environments with dependencies without Internet connection.
+
+**Not in scope**
+* Setup additional infrastructure to support disconnected environments (eg. private registries).
+
+**Future scope**
+* Package dependencies selectively.
+* Populate disconnected environments with dependencies with Internet connection.
+ 
+## Current state
+In order to support disconnected environments, users currently have to:
+1. Provision an OCI registry.
+1. Run `eksctl anywhere download artifacts` to download manifests to disk.
+1. Run `eksctl anywhere download images` to download images and charts to disk.
+1. Run `eksctl anywhere import images` to push container images and charts to a registry.
+1. Configure their cluster spec file to point to the registry 
+1. Run all cluster commands (`create`, `upgrade`, etc.) with the `--bundles-override` flag pointing to the `Bundles` manifest downloaded in the first step.
+
+## Overview of Solution
+1. Users will provide an OCI registry.
+1. We will store all dependencies (container images, helm charts and yaml manifests) in that registry.
+1. If a cluster config with a [`registryMirrorConfiguration`](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L51) is provided, the CLI and the rest of components will always pull them from it.
+No extra flags/configuration will be needed and the user will interact with this cluster as with any other one.
+
+## Solution Details
+
+### Populating the registry
+
+In order to populate the registry, we will offer two commands:
+```sh
+eksctl anywhere export artifacts --output artifacts.tar
+eksctl anywhere import artifacts --input artifacts.tar --registry myregistry.com
+```
+
+The first one downloads all dependencies (yaml manifests, images and helm charts). The only command argument here is the destination file. This command will create a tarball containing all 3 types of artifacts in that location.
+The second one unpackages the tarball created in the first command, reads the packaged `Bundles` manifest and imports the referenced dependencies to a registry. The input here will be the artifacts tarball, the registry endpoint and the registry credentials (provided through env vars).
```

Comment:
What do you mean by validating? Checking that all the artifacts referenced in the `Bundle` are present in the tarball? That might not be feasible for container images, since they are all packaged together. What we can validate is that all referenced images end up in the registry. But that validation is kind of implied since the `push` action would fail in case one image is not present.

What I was thinking is that I might backtrack on reading from the `Bundles` and just include an extra "artifacts list" in the tarball, referencing all included artifacts. That would allow us to add options to selectively include/exclude artifacts in the future.

Reply:
Yea I was more referring to running import based on what is referenced in `Bundle` and ensuring that we can find it in the tarball. 

Another option is that we have the tarball created as part of our build infra and just download that? Just throwing ideas.

Reply:
Yeah that's what I was thinking. But we could leave this decision for the implementation. Another option more future proof is to include a "artifact manifest" in the tarball that references all included artifacts. That makes it more flexible if one day we want to offer "selectively including artifacts". Like opting out from some provider images or opt in for curated packages.

About prebuilding the tarball and offering it to the public, yeah agreed, that's what I proposed in the `Pre packaged dependencies` section.